packages/server/graphql/mutations/resetMeetingToStage.ts

```diff
@@ -0,0 +1,134 @@
+      return standardError(new Error('Not meeting facilitator anymore'), {userId: viewerId})
+    }
+
+    // VALIDATION
+    const foundResponse = findStageById(phases, stageId)
+    if (!foundResponse)
+      return standardError(new Error('Meeting stage not found'), {userId: viewerId})
+    const {stage: resetToStage} = foundResponse!
+    if (!resetToStage.isNavigableByFacilitator)
+      return standardError(new Error('Stage has not started'), {userId: viewerId})
+    if (!resetToStage.isComplete)
+      return standardError(new Error('Stage has not finished'), {userId: viewerId})
+    if (resetToStage.phaseType !== NewMeetingPhaseTypeEnum.group)
+      return standardError(new Error('Resetting to this stage type is not supported'), {
+        userId: viewerId
+      })
+    if (meeting.endedAt)
+      return standardError(new Error('The meeting has already ended'), {userId: viewerId})
+
+    // RESOLUTION
+    const meetingCount = await r
+      .table('NewMeeting')
+      .getAll(teamId, {index: 'teamId'})
+      .filter({meetingType})
+      .count()
+      .default(0)
+      .run()
+    const createdPhases = await createNewMeetingPhases(
+      teamId,
+      meetingCount,
+      meetingType,
+      dataLoader
+    )
+    let shouldResetStage = false
+    let resetToPhaseIndex = -1
+    const newPhases = [] as GenericMeetingPhase[]
+    for (const [phaseIndex, phase] of phases.entries()) {
+      const {stages} = phase
+      if (!stages) continue
+      const newStages = [] as GenericMeetingStage[]
+      for (const [stageIndex, stage] of stages.entries()) {
+        if (stage.id === stageId) {
+          shouldResetStage = true
+          resetToPhaseIndex = phaseIndex
+        }
+        if (!shouldResetStage) {
+          newStages.push(stage)
+          continue
+        }
+        const resettedStage = createdPhases[phaseIndex]?.stages[stageIndex]
+        if (!resettedStage) continue
+        newStages.push(Object.assign(resettedStage, {id: stage.id}))
+      }
+      phase.stages = newStages
+      newPhases.push(phase)
+    }
+    primePhases(newPhases, resetToPhaseIndex)
+
+    const reflectionGroups = await dataLoader
+      .get('retroReflectionGroupsByMeetingId')
+      .load(meetingId)
+    const reflectionGroupIds = reflectionGroups.map((rg) => rg.id)
+
+    await Promise.all([
+      r
+        .table('Comment')
+        .getAll(r.args(reflectionGroupIds), {index: 'threadId'})
+        .delete()
+        .run(),
+      r
+        .table('Task')
+        .getAll(r.args(reflectionGroupIds), {index: 'threadId'})
+        .delete()
+        .run(),
+      r
+        .table('RetroReflectionGroup')
+        .getAll(r.args(reflectionGroupIds))
+        .update({voterIds: []})
+        .run(),
+      r
+        .table('NewMeeting')
+        .get(meetingId)
+        .update({phases: newPhases})
+        .run(),
+      (r.table('MeetingMember').getAll(meetingId, {index: 'meetingId'}) as any)
```

Comment:
nit: can cast to `RetroMeetingMember` to get free typechecking of the votesRemaining field

Reply:
true! out of curiosity and if anything comes top of mind, how would you handle `Property 'update' does not exist on type 'RetroMeetingMember'.`?